releases/2022-11-25-releasenotes.md

```diff
@@ -0,0 +1,154 @@
+tags:
+- Improvements
+- Developer Experience
+- GraphQL
+---
+
+Vendia Engineers worked tirelessly to improve our product and ensure that we follow and set best practices for industrial standards. Here are the newest updates!
+
+# Error handling the way it should with [GraphQL](https://graphql.org/). 
+
+Errors are not scary as long as we handle them well. In the past Vendia's end point would return 4XX and 5XX status codes for any http request failures. But that has changed as we decided to follow GraphQL's [recommended approach](https://graphql.github.io/graphql-over-http/draft/#sec-Field-errors-encountered-during-execution) over http calls.
+
+In short, it means the end point will always return 2XX. But when error happens, error messages will be included in the response. 
+
+Here's two examples using `Curl`:
+
+1. A request with typo in it's query. `n` is intentionally removed in `listVendia_BlockItems`:
+```
+curl 'https://<some-domain>/graphql/' \
+    -H 'Authorization: <api-token>' \
+    -H 'content-type: application/json' \
+    -H 'accept: application/json' \
+    -i \
+    --data-raw '{"query":"query blocksQuery {  listVedia_BlockItems {    Vendia_BlockItems {      blockId      blockHash    }  }}","variables":null,"operationName":"blocksQuery"}'
+```
+
+We can see that it returns `200` status code. But also gives detailed error messages pointing out that `listVedia_BlockItems` is incorrect and suggested other available options.
+
+Sample Response:
+```
+HTTP/2 200
+date: Tue, 29 Nov 2022 20:31:27 GMT
+content-type: application/json
+content-length: 288
+x-amzn-requestid: XXXXXX
+access-control-allow-origin: *
+x-amz-apigw-id: XXXXXX
+x-amzn-trace-id: XXXXXX
+
+{"data": null, "errors": [{"message": "Cannot query field 'listVedia_BlockItems' on type 'Query'. Did you mean 'listVendia_BlockItems', 'listVendia_FileItems', 'listVendia_FolderItems', 'listVendia_ApiKeyItems', or 'listVendia_ContractItems'?", "locations": [{"line": 1, "column": 22}]}]}%
+```
+
+2. A request with meaningless query body:
+
+```
+curl 'https://<some-domain>/graphql/' \
+    -H 'Authorization: <api-token>' \
+    -H 'content-type: application/json' \
+    -H 'accept: application/json' \
+    -i \
+    --data-raw '{"query":"query "}'
+```
+
+Again, we see a `200` status code. The message informs us that the syntax is not right which makes sense since our query body has nothing.
+
+Sample Response:
+```
+HTTP/2 200
+date: Tue, 29 Nov 2022 20:31:52 GMT
+content-type: application/json
+content-length: 124
+x-amzn-requestid: 6097318d-f50d-4097-b415-82e6a8f07652
+access-control-allow-origin: *
+x-amz-apigw-id: cYXYzEpbyK4FsBg=
+x-amzn-trace-id: Root=1-63866c38-38b964043a2d77c240da4981;Sampled=0
+
+{"data": null, "errors": [{"message": "Syntax Error: Expected '{', found <EOF>.", "locations": [{"line": 1, "column": 7}]}]}%
+```
+
+It is recommended to understand the [common GraphQL error handling techniques](https://the-guild.dev/blog/graphql-error-handling-with-fp).
```

Comment:
Is this link a reference for customers to see an approach to consume the error messages we present and process retries or exceptions accordingly?  Or is it more just context on how we may have implemented error handling?   

- If the former - that's great! And I'd call that out, something like "For examples on error handling, see {link}".  

- If it's the latter, we can probably exclude it as it may be too detailed but I'll defer to what others think if they think customers will like this depth. If it is, a phrasing suggestion to: "For more information on building error handling solutions, see {link}".

Reply:
it's former.